lib/matplotlib/figure.py

```diff
@@ -290,6 +290,69 @@ def set_figure(self, fig):
                       doc=("The root `Figure`.  To get the parent of a `SubFigure`, "
                            "use the `get_figure` method."))
 
+    def set_subplotparams(self, subplotparams={}):
+        """
+        Set the subplot layout parameters.
+        Accepts either a `.SubplotParams` object, from which the relevant
+        parameters are copied, or a dictionary of subplot layout parameters.
+        If a dictionary is provided, this function is a convenience wrapper for
+        `matplotlib.figure.Figure.subplots_adjust`
+
+        Parameters
+        ----------
+        subplotparams : `~matplotlib.figure.SubplotParams` or dict with keys
+            "left", "bottom", "right", 'top", "wspace", "hspace"] , optional
+            SubplotParams object to copy new subplot parameters from, or a dict
+            of SubplotParams constructor arguments.
+            By default, an empty dictionary is passed, which maintains the
+            current state of the figure's `.SubplotParams`
+
+        See Also
+        --------
+        matplotlib.figure.Figure.subplots_adjust
+        matplotlib.figure.Figure.get_subplotparams
+        """
+
+        subplotparams_args = ["left", "bottom", "right",
+                              "top", "wspace", "hspace"]
+        kwargs = {}
+        if isinstance(subplotparams, SubplotParams):
+            for key in subplotparams_args:
+                kwargs[key] = getattr(subplotparams, key)
+        elif isinstance(subplotparams, dict):
+            for key in subplotparams.keys():
+                if key in subplotparams_args:
+                    kwargs[key] = subplotparams[key]
+                else:
+                    _api.warn_external(
+                        f"'{key}' is not a valid key for set_subplotparams;"
+                        " this key was ignored.")
+        else:
+            raise TypeError(
+                "subplotparams must be a dictionary of keyword-argument pairs or"
+                " an instance of SubplotParams()")
+        if kwargs == {}:
+            self.set_subplotparams(self.get_subplotparams())
+        self.subplots_adjust(**kwargs)
+
+    def get_subplotparams(self):
```

Comment:
This method is not run by the tests because you replaced it in the `Figure` subclass, so only the subclass one gets run by your new test.  I'm not too familiar with subplotparams but, as far as I can see, they only exist on a `Figure` and not on a `SubFigure` so both these methods could go directly on the `Figure` class.

Reply:
@rcomer Fixed the above inheritance issue. Thanks for pointing it out. I've moved the definitions to Figure and all the pytest cases passed locally. Could you please review my failing pytest macOS tests?

Reply:
Those MacOS ones are often flakey.  I’ve re-started them to see if they do better this time 🤞

Reply:
For the documentation failure I _think_ you just need to add your new methods to an appropriate section here, to make them show up in the docs:
https://github.com/matplotlib/matplotlib/blob/main/doc/api/figure_api.rst